src/app/body/milestone-details/milestone-details.component.ts

```diff
@@ -78,6 +125,70 @@ export class MilestoneDetailsComponent implements OnInit {
         }
       });
   }
+  activateAdd(){
+  this.addTaskActive = true;
+  this.getAllTasks();
+  }
+
+  getAllTasks(){
+    this.showLoader = true;
+    const callable = this.functions.httpsCallable("tasks/getAllTasks");
+
+    callable({ OrgDomain: this.orgDomain, TeamId: this.milestoneData.TeamId }).pipe(
+      map(actions => {
+        return actions.data as Tasks[];
+      })).subscribe({
+        next: (data) => {
+          const temp = data;
+          this.allTasks = temp.filter((element)=>{
+            if(element.MilestoneId != this.milestoneId){
+              return element;
+            }
+          });
+        },
+        error: (error) => {
+          console.log(error);
+        },
+        complete: () => {
+          console.info("Fetched All Tasks Successfully");
+          this.taskDataReady = true;
+          this.showLoader = false;
+        }
+      });
+  }
+  setMilestoneWidth = function () {
+    //Calculations are adjusted for UI Improvisations
+    var width = (100 / (this.taskData.length)).toString() + "%";
+    return width;
+  };
+  setProgressWidth() {
+    //Calculations are adjusted for UI Improvisations
+    var width = ((((this.totalCompletedTasks - 1) / (this.totalTasks)) * 100)+3).toString() + "%";
+    return width;
+  }
+
+ async addTask(task){
+    this.showLoader = true
+    const todayDate = this.toolsService.date();
+    const time = this.toolsService.time();
+    const appKey = this.backendService.getOrganizationAppKey();
+      const callable = this.functions.httpsCallable('tasks/editTask');
+      await callable({Title: task.Title, Status: task.Status, AppKey: appKey, Id: task.Id, Description: task.Description, Priority: task.Priority, Difficulty: task.Difficulty, Assignee: task.Assignee, EstimatedTime: task.EstimatedTime, Project: task.Project, SprintNumber: task.SprintNumber, StoryPointNumber: task.StoryPointNumber, OldStoryPointNumber: task.StoryPointNumber, PreviousId: task.SprintNumber, CreationDate: task.CreationDate, Date: todayDate, Time: time, ChangedData: "Milestone Added", Uid: this.authService.user.uid, Type:task.Type, Reporter: task.reporterName, MilestoneId: this.milestoneId}).subscribe({
```

Comment:
remove async and await, it's not needed when subscribe is used. do this everywhere.

Reply:
Done